src/interface/state_descriptor.hpp

```diff
@@ -49,8 +49,11 @@ class StateDescriptor {
   explicit StateDescriptor(std::string const &label) : label_(label) {}
 
   template <typename T>
-  void AddParam(const std::string &key, T value) {
+  void AddParam(const std::string &key, T value, const bool outputing = false) {
```

Comment:
The name `outputing` feels odd here. Why not just `output`?

Reply:
I used `outputing` to match the `outputing_params` vector that I've added. `OutputParams` is an entirely separate object, so I avoided using `output` as an verb here to avoid confusion. Would `include_with_output` be too verbose or should I just use `output`?

Reply:
I like `include_with_output`